pkg/deploy/generator/scripts/util-system.sh

```diff
@@ -0,0 +1,300 @@
+    rotate 1
+}'
+
+    write_file logrotate_conf_filename logrotate_conf_file true
+
+    if [ -n "${dropin_files[*]}" ]; then
+        local -r logrotate_d="/etc/logrotate.d"
+        log "Writing logrotate files to $logrotate_d"
+        for dropin_name in "${!dropin_files[@]}"; do
+            # shellcheck disable=SC2034
+            local -r dropin_filename="$logrotate_d/$dropin_name"
+            # shellcheck disable=SC2034
+            local -r dropin_file="${dropin_files["$dropin_name"]}"
+            write_file dropin_filename dropin_file true
+        done
+    fi
+}
+
+# pull_container_images
+# args:
+# 1) pull_images - nameref, string array
+# 2) registry_conf - nameref, string, optional; path to docker/podman configuration file
+pull_container_images() {
+    local -n pull_images="$1"
+    local -n registry_conf="${2:-empty_str}"
+    log "starting"
+
+    # shellcheck disable=SC2034
+    local -ri retry_time=30
+    # The managed identity that the VM runs as only has a single roleassignment.
+    # This role assignment is ACRPull which is not necessarily present in the
+    # subscription we're deploying into.  If the identity does not have any
+    # role assignments scoped on the subscription we're deploying into, it will
+    # not show on az login -i, which is why the below line is commented.
+    # az account set -s "$SUBSCRIPTIONID"
+    cmd=(
+        az
+        login
+        -i
+        --allow-no-subscriptions
+    )
+
+    log "Running az login with retries"
+    retry cmd retry_time
+
+    # Suppress emulation output for podman instead of docker for az acr compatability
+    mkdir -p /etc/containers/
+    mkdir -p /root/.docker
+    touch /etc/containers/nodocker
+
+    # This name is used in the case that az acr login searches for this in it's environment
+    export REGISTRY_AUTH_FILE="/root/.docker/config.json"
+    
+    if [ -n "${registry_conf}" ]; then
+        write_file REGISTRY_AUTH_FILE registry_conf true
+    fi
+
+    log "logging into prod acr"
+    cmd=(
+        az
+        acr
+        login
+        --name
+        # TODO replace this with variable expansion
+        # Reference: https://www.shellcheck.net/wiki/SC2001
+        "$(sed -e 's|.*/||' <<<"$ACRRESOURCEID")"
+    )
+
+    retry cmd retry_time
+
+    # shellcheck disable=SC2068
+    for i in ${pull_images[@]}; do
+        local -n image="$i"
+        cmd=(
+            podman
+            pull
+            "$image"
+        )
+
+        log "Pulling image $image with retries now"
+        retry cmd retry_time
+    done
+
+    # shellcheck disable=SC2034
+    cmd=(
+        az
+        logout
+    )
+
+    log "Running az logout with retries"
+    retry cmd retry_time
+}
+
+# configure_certs
+# args:
+# 1) role - string; can be "devproxy" or "rp"
+configure_certs() {
+    local -n role="$1"
+    log "starting"
+    log "Configuring certificates for $role"
+
+    verify_role role true
+
+    if [ "$role" == "devproxy" ]; then
```

Comment:
Nit: refactor code so we have `configure_devproxy_certs` and `configure_rp_certs` to avoid flag arguments.